pkg/restore/restore.go

```diff
@@ -474,14 +485,33 @@ func (ctx *context) restoreFromDir(dir string) (api.RestoreResult, api.RestoreRe
 			if !existingNamespaces.Has(mappedNsName) {
 				logger := ctx.log.WithField("namespace", nsName)
 				ns := getNamespace(logger, filepath.Join(dir, api.ResourcesDir, "namespaces", api.ClusterScopedDir, nsName+".json"), mappedNsName)
-				if _, err := kube.EnsureNamespaceExists(ns, ctx.namespaceClient); err != nil {
+
+				clusterNS, err := ctx.namespaceClient.Get(mappedNsName, metav1.GetOptions{})
+				if err != nil && !apierrors.IsNotFound(err) {
 					addArkError(&errs, err)
 					continue
+				} else if apierrors.IsNotFound(err) {
+					_, err := ctx.namespaceClient.Create(ns)
+					if err != nil {
+						addArkError(&errs, err)
+						continue
+					}
 				}
 
+				// Namespace exists but is terminating
+				if clusterNS != nil && clusterNS.Status.Phase == v1.NamespaceTerminating {
```

Comment:
You shouldn't need to check for `clusterNS != nil` - if you make it here, it means `ctx.namespaceClient.Get()` returned no error.